### ISSUE TITLE
SugarSync broke for me in release 1.56.0

### ISSUE THREAD

**Issue by pcsanza** [rclone/rclone#5525]
#### Output of `rclone version`
```
rclone v1.56.0
- os/version: ubuntu 20.04 (64 bit)
- os/kernel: 5.4.0-80-generic (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.16.5
- go/linking: static
- go/tags: none
```

#### Describe the issue
Trying to set up rclone for a SugarSync remote on my Linux box (Ubuntu 20.04 64-bit).  
Running `rclone config` and selecting SugarSync (# 33 in 1.56.0) and accepting all defaults and supplying my sugarsync username (email) and password when prompted results in an error message 
`Error: failed to get token: HTTP error 401 (401 Unauthorized)`

I created a new question on the forum here....
[https://forum.rclone.org/t/trouble-getting-sugarsync-remote-configured/25864](url)

A very helpful person finally confirmed that a sugarsync remote could not be created successfully using rclone 1.56.0 but when rclone was rolled back to 1.55.1, then a sugarsync remote could be created successfully.  I have not independently confirmed this result yet but plan to as soon as I can figure out how to roll back my rclone.

**Comment by ncw** [comment]
We re-did the config mechanism for all the backends for 1.56.

It looks like sugarsync got broken.

If you create the remote with 1.55.1 it will work with 1.56 - I have a working remote with 1.56 that way.

Here are the changes from 1.55 -> 1.56

    $ git log --oneline v1.55.0.. -- backend/sugarsync/

- f122808d8 fs: add names to each config parameter so we can override them #3455
- 94dbfa4ea fs: change Config callback into state based callback #3455
- b78c9a65f backends: remove log.Fatal and replace with error returns #5234

I've had a go at fixing this - can you try this please?

[v1.57.0-beta.5614.5e64364aa.fix-5525-sugarsync](https://beta.rclone.org/branch/fix-5525-sugarsync/v1.57.0-beta.5614.5e64364aa.fix-5525-sugarsync/) on branch [fix-5525-sugarsync](https://github.com/rclone/rclone/tree/fix-5525-sugarsync) (uploaded in 15-30 mins)

**Comment by pcsanza** [comment]
Hi - Sorry about the delayed response - day job....
I tried downloading
https://beta.rclone.org/branch/fix-5525-sugarsync/v1.57.0-beta.5614.5e64364aa.fix-5525-sugarsync/rclone-v1.57.0-beta.5614.5e64364aa.fix-5525-sugarsync-linux-amd64.zip
but it appears to be empty....  Tried several others in the same
fix-5525-sugarsync folder and they all appear empty.

**Comment by pcsanza** [comment]
Strangely enough, the windows amd64 zip is there and downloadable but the
linux amd64 zip is not.....

**Comment by ncw** [comment]
That download is working for me now. Can you give it another go?

**Comment by pcsanza** [comment]
Hey - thanks for jumping on this!  I managed to sneak a connection to the
Internet and try it out....  Yes, 1.57.0-beta with fix-5525-sugarsync
allowed me to create a new remote for sugarsync and access files/folders on
it.  Will you mark the issue closed or do I need to do something on
github?

Thanks again.


-Peter

**Comment by ncw** [comment]
Thanks for testing

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.57 and 1.56.1
